test/extended/util/disruption/disruption.go

```diff
@@ -280,9 +280,21 @@ func createTestFrameworks(tests []upgrades.Test) map[string]*framework.Framework
 // disruption flake if any disruption occurs, and uses reason to prefix the message. I.e. tolerate 0.1 of 10m total will fail
 // if the sum of the intervals is greater than 1m, or report a flake if any interval is found.
 func ExpectNoDisruption(f *framework.Framework, tolerate float64, total time.Duration, events monitor.EventIntervals, reason string) {
+	duration, describe := GetDisruption(events, "")
+	if percent := float64(duration) / float64(total); percent > tolerate {
+		framework.Failf("%s for at least %s of %s (%0.0f%%):\n\n%s", reason, duration.Truncate(time.Second), total.Truncate(time.Second), percent*100, strings.Join(describe, "\n"))
+	} else if duration > 0 {
+		FrameworkFlakef(f, "%s for at least %s of %s (%0.0f%%), this is currently sufficient to pass the test/job but not considered completely correct:\n\n%s", reason, duration.Truncate(time.Second), total.Truncate(time.Second), percent*100, strings.Join(describe, "\n"))
+	}
+}
+
+func GetDisruption(events monitor.EventIntervals, requiredLocator string) (time.Duration, []string) {
```

Comment:
I haven't looked at this before.  Can you provide an example eventInterval that shows what you're looking for in this list?

Reply:
fixed

Reply:
EventInterval is basically what's printed in `Monitor cluster while tests execute output`:

```
Oct 12 15:24:26.863 E kube-apiserver Kube API started failing: namespaces "kube-system" not found
Oct 12 15:24:27.759 - 5s    E kube-apiserver Kube API is not responding to GET requests
Oct 12 15:24:33.861 I kube-apiserver Kube API started responding to GET requests
```

Showing "From" above as the timestamp. Most EventIntervals have `From == To` (== when the event happened), but `kube-apiserver Kube API is not responding to GET requests` has almost the right `From`/`To`  (both +/- 1s) for total duration of the outage.